### PR TITLE
chore(test): another mint module recovery test

### PR DESCRIPTION
**Pull request by dpc**

### CODE REVIEW

**Review by elsirion — COMMENTED** [review]
Is there a way to do this as an in-process/unit test? These CLI tests are very limited in their ability to explore weird scenarios that might only happen once in a blue moon in practice.

**Comment by dpc** [comment]
@elsirion Maybe, but would need a description of a guess of a scenario that we're missing here.

The recovery logic does not have a lot of weird cases in my mind. You start from the empty state and rewind existing history. Any state that the user can hit, we should be able to hit with just the cli.

File: modules/fedimint-mint-tests/src/bin/mint-client-restore.rs [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    devimint::run_devfed_test(|fed, _process_mgr| async move {
+        let fed = fed.fed().await?;
+
+        test_restore_gap_test(fed).await?;
+        Ok(())
+    })
+    .await
+}
+
+pub async fn test_restore_gap_test(fed: &Federation) -> Result<()> {
```
Comment (elsirion) [inline modules/fedimint-mint-tests/src/bin/mint-client-restore.rs:21]: It says something about the gap limit, but it's unclear to me which guarantees about preconditions the code provides. The coin selection and reissuance is opaque, so we don't really know which scenario will be tested. Ideally we could test scenarios where transactions failed and e-cash outputs were cancelled, producing holes in the used indexes.

Comment (dpc) [reply modules/fedimint-mint-tests/src/bin/mint-client-restore.rs:21]: @elsirion So your worry is that on failed paymenets client allocates the note_idx, but never uses it, potentially creating holes of unused mode_idx larger than gap limit? In that case why don't we just increase the gap limit? I mean - there is no other way anyway.

Comment (elsirion) [reply modules/fedimint-mint-tests/src/bin/mint-client-restore.rs:21]: Yeah, that came to my mind too. I've been staring at the mint recovery code all afternoon and haven't seen anything suspicious.

Comment (elsirion) [reply modules/fedimint-mint-tests/src/bin/mint-client-restore.rs:21]: @dpc https://github.com/fedimint/fedimint/pull/6096

Comment (dpc) [reply modules/fedimint-mint-tests/src/bin/mint-client-restore.rs:21]: @elsirion Sir, I can test that to, see last commit.

Comment (dpc) [reply modules/fedimint-mint-tests/src/bin/mint-client-restore.rs:21]: I think we've found something.

Comment (dpc) [reply modules/fedimint-mint-tests/src/bin/mint-client-restore.rs:21]: @elsirion Actually, it was just artifact of multiple gaps adding up and crossing the gap limit, it seems. Everything seems to be working.

**Review by elsirion — APPROVED** [review]

**Comment by elsirion** [comment]
More tests more good, but I wish this was a unit test. You built a beautifully testable recovery logic, that is very self-contained at its core and only relies on DB access at the very end. These CLI tests are really hard to debug should they ever become flaky.

**Pull request merged**